lib/internal/Magento/Framework/Mview/View.php

```diff
@@ -277,40 +277,51 @@ public function update()
             try {
                 $this->getState()->setStatus(View\StateInterface::STATUS_WORKING)->save();
 
-                $versionBatchSize = self::$maxVersionQueryBatch;
-                $batchSize = isset($this->changelogBatchSize[$this->getChangelog()->getViewId()])
-                    ? $this->changelogBatchSize[$this->getChangelog()->getViewId()]
-                    : self::DEFAULT_BATCH_SIZE;
-
-                for ($vsFrom = $lastVersionId; $vsFrom < $currentVersionId; $vsFrom += $versionBatchSize) {
-                    // Don't go past the current version for atomicy.
-                    $versionTo = min($currentVersionId, $vsFrom + $versionBatchSize);
-                    $ids = $this->getChangelog()->getList($vsFrom, $versionTo);
-
-                    // We run the actual indexer in batches.
-                    // Chunked AFTER loading to avoid duplicates in separate chunks.
-                    $chunks = array_chunk($ids, $batchSize);
-                    foreach ($chunks as $ids) {
-                        $action->execute($ids);
-                    }
-                }
+                $this->executeAction($action, $currentVersionId, $lastVersionId);
 
                 $this->getState()->loadByView($this->getId());
                 $statusToRestore = $this->getState()->getStatus() == View\StateInterface::STATUS_SUSPENDED
                     ? View\StateInterface::STATUS_SUSPENDED
                     : View\StateInterface::STATUS_IDLE;
                 $this->getState()->setVersionId($currentVersionId)->setStatus($statusToRestore)->save();
-            } catch (\Exception $exception) {
+            } catch (\Throwable $exception) {
                 $this->getState()->loadByView($this->getId());
                 $statusToRestore = $this->getState()->getStatus() == View\StateInterface::STATUS_SUSPENDED
                     ? View\StateInterface::STATUS_SUSPENDED
                     : View\StateInterface::STATUS_IDLE;
                 $this->getState()->setStatus($statusToRestore)->save();
+                if (!$exception instanceof \Exception) {
+                    $exception = new \RuntimeException(
+                        'Error when updating an mview',
+                        0,
+                        $exception
+                    );
+                }
                 throw $exception;
             }
         }
     }
 
+    private function executeAction($action, $currentVersionId, $lastVersionId) {
```

Comment:
Typehints, please, and declare strict types should also work, I believe.

Reply:
@orlangur Done

Reply:
@QuentinFarizonAfrimarket please add `declare(strict types)` on top and if builds will be green - squash into a single commit. If not, don't add such line and just squash.

Reply:
Done, tests pass with strict type